### PR TITLE
breaking: rename numpy_to_record_serializer to RecordSerializer

### PR DESCRIPTION
**Pull request by bveeramani**
*Issue #, if available:*

*Description of changes:*
Renamed numpy_to_record_serializer to RecordSerializer.

*Testing done:*
Fixed relevant tests.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/amazon/common.py [hunk 1]
```diff
@@ -22,31 +22,35 @@
...
-        assert len(array.shape) == 2, "Expecting a 1 or 2 dimensional array"
-        buf = io.BytesIO()
-        write_numpy_to_dense_tensor(buf, array)
-        buf.seek(0)
-        return buf
+        if len(data.shape) == 1:
+            data = data.reshape(1, data.shape[0])
+
+        if len(data.shape) != 2:
+            raise ValueError("Expected a one or two dimensional array.")
```
Comment (laurenyu) [inline src/sagemaker/amazon/common.py:47]: it might be helpful to include information about the user input in the error message

Comment (bveeramani) [reply src/sagemaker/amazon/common.py]: Fixed

File: src/sagemaker/amazon/common.py [hunk 2]
```diff
@@ -22,31 +22,37 @@
...
 from sagemaker.amazon.record_pb2 import Record
 from sagemaker.deserializers import BaseDeserializer
+from sagemaker.serializers import BaseSerializer
 from sagemaker.utils import DeferredError
 
 
-class numpy_to_record_serializer(object):
-    """Placeholder docstring"""
+class RecordSerializer(BaseSerializer):
+    """Serialize array-like data for an inference request."""
```
Comment (laurenyu) [inline src/sagemaker/amazon/common.py:30]: nit: might want to be more specific that this is for numpy arrays

File: src/sagemaker/amazon/common.py [hunk 3]
```diff
@@ -22,31 +22,37 @@
...
-    def __init__(self, content_type="application/x-recordio-protobuf"):
-        """
-        Args:
-            content_type:
-        """
-        self.content_type = content_type
+    CONTENT_TYPE = "application/x-recordio-protobuf"
+
+    def serialize(self, data):
+        """Serialize array-like data into a buffer containing RecordIO records.
```
Comment (laurenyu) [inline src/sagemaker/amazon/common.py:35]: same here

File: src/sagemaker/amazon/common.py [hunk 4]
```diff
@@ -22,31 +22,37 @@
...
 
-    def __call__(self, array):
-        """
         Args:
-            array:
+            data (numpy.ndarray): The data to serialize.
+
+        Returns:
+            _io.BytesIO: A buffer containing the data serialized as records.
```
Comment (laurenyu) [inline src/sagemaker/amazon/common.py:41]:
```suggestion
            io.BytesIO: A buffer containing the data serialized as records.
```

**Pull request merged**